src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -41,7 +41,7 @@
 
     private Map<HttpConnection, SourceBridgeEndpoint> httpSourceEndpoints;
 
-    private Map<String, SinkBridgeEndpoint> httpSinkEndpoints;
+    protected static Map<String, SinkBridgeEndpoint> httpSinkEndpoints;
```

Comment:
Instead of exposing as static the endpoints collection to ... an HTTP endpoint itself (where you check if the consumerId already exists), we should do the check in the HTTP bridge as we do for the `DELETE` operation.

Reply:
Offline you noticed that the `name` comes from the body not URL as for `DELETE` so to avoid this `static` field I would ...

* create an `HttpBridgeContext` for holding the source and sink endpoints
* allowing to pass the context to the `HttpSinkBridgeEndpoint` in the ctor so that it can access to the sinks and checking for existing consumer-id

Wdyt?

Reply:
I can do that.